### PR TITLE
DEX-366 using the ia config for the detection

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Using the ia config for detection
- Tests updated to do so.
- asset_group model config given default and made non nullable as it was causing problems.
- Tests updated to clear up asset group on failure (not all failures were handled correctly)

---

### CODE REVIEW

File: app/modules/asset_groups/models.py [hunk 1]
```diff
@@ -558,7 +555,7 @@ class AssetGroup(db.Model, HoustonModel):
 
     description = db.Column(db.String(length=255), nullable=True)
 
-    config = db.Column(db.JSON, nullable=True)
+    config = db.Column(db.JSON, default=lambda: {}, nullable=False)
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:558]: Leaving this potentially nullable was causing issues. Safer to have it as an empty dict.

File: tests/modules/asset_groups/test_models.py [hunk 2]
```diff
@@ -45,25 +45,26 @@ def test_asset_group_sightings_jobs(flask_app, db, admin_user, test_root, reques
     uuids = [job_id2, job_id1]
 
     # Don't send anything to acm
-    with mock.patch('app.modules.asset_groups.models.current_app'):
+    with mock.patch('app.modules.asset_groups.models.current_app') as mock_app:
+        mock_app.config.get.return_value = 'zebra'
```
Comment (hwindsor) [inline tests/modules/asset_groups/test_models.py:49]: This is actually the get for 'CONFIG_MODEL' Should really make sure it's just that.

**Comment by colinwkingen** [comment]
Looks like a couple conflicts in here @hwindsor , but I've read the code and think you are on the right track.

**Review by colinwkingen — APPROVED** [review]
LGTM and will merge after conflict resolution.

**Pull request merged**